addons/stock_picking_batch/models/stock_picking_batch.py

```diff
@@ -91,6 +94,30 @@ def _compute_show_validate(self):
         for batch in self:
             batch.show_validate = any(picking.show_validate for picking in batch.picking_ids)
 
+    @api.depends('state', 'move_ids')
+    def _compute_show_allocation(self):
+        self.show_allocation = False
+        if not self.user_has_groups('stock.group_reception_report'):
```

Comment:
Can you create a method on the picking model shared by batch picking

```python
_get_show_allocation(self, moves, picking_type_id):
     if not picking_type_id or picking_type_id.code == 'outgoing':
          return False
     lines = moves.filtered(lambda m: m.product_id.type == 'product' and m.state != 'cancel')
     ... 
           ('picking_id', 'not in', self.ids)
```
in picking:
```python
for picking in self:
     picking.show_allocation = picking._get_show_allocation(picking.move_lines, picking.picking_type_id
```

in batch:
```python
for batch in self:
     batch.show_allocation = batch.picking_ids._get_show_allocation(picking.move_lines, picking.picking_type_id
```